tests/integration_tests/modules/test_set_hostname.py

```diff
@@ -30,35 +30,100 @@
 fqdn: cloudinit2.test.io
 """
 
+REQUIRES_FILE_FLAG = PLATFORM == "gce" and CURRENT_RELEASE >= NOBLE
+
 
 @pytest.mark.ci
 class TestHostname:
     @pytest.mark.user_data(USER_DATA_HOSTNAME)
     def test_hostname(self, client):
         hostname_output = client.execute("hostname")
-        assert "cloudinit2" in hostname_output.strip()
+        if REQUIRES_FILE_FLAG:
+            assert "cloudinit2" not in hostname_output.strip()
```

Comment:
How are the instances created?  I ask because the randomly assigned google DHCP hostname could include cloudinit2 if the instance name includes that.

Reply:
In integration tests, the instance name will never be that unless you very intentionally instrument it to happen. Instance names should have a 'cloud-init' + timestamp in the name.

Reply:
@catmsred from cloud-init's integration tests, [the launched instance name defined at launch time here](https://github.com/canonical/cloud-init/blob/main/tests/integration_testshe/clouds.py#L333)